ui/plugin-system/src/components/PluginEditor/plugin-editor-api.ts

```diff
@@ -81,7 +100,12 @@ export function usePluginEditor(props: UsePluginEditorProps) {
   // When kind changes and we haven't loaded that plugin before, we will need to enter a "pending" state so that we
   // can generate proper initial spec values that match the new plugin kind
   const [pendingKind, setPendingKind] = useState(initPendingKind);
-  const { data: plugin, isFetching, error } = usePlugin(pluginType, pendingKind);
+  console.log('before');
+  console.log(pluginType, pendingKind);
+  const { data: listPluginMetadata } = useListPluginMetadata(pluginTypes);
```

Comment:
Reusing the list plugin metadata to find back the pluginType from kind every time the kind is changed. This might be not ideal but it works and I wanted to have something that I could display hehe